### PR TITLE
Mr/create posts

### PR DESCRIPTION
**Pull request by mronauli**
## Description
Add functionality for user to create new post

## Notes
Thinking I may be way off the mark in terms of testing... interested to hear everyone's thoughts

## RSpec Results
```
Finished in 1.33 seconds (files took 3.87 seconds to load)
40 examples, 0 failures

Coverage report generated for RSpec to /Users/mariaronauli/turing/3module/projects/ghost-diary-gateway/coverage. 440 / 440 LOC (100.0%) covered.
```

### CODE REVIEW

File: spec/requests/profile/posts/new_spec.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require "rails_helper"
```
Comment (d-atkins) [inline spec/requests/profile/posts/new_spec.rb:1]: There is no API exposure in the gateway so this test should not be a request test.

File: app/controllers/profile/posts_controller.rb [hunk 2]
```diff
@@ -8,4 +8,15 @@ def show
     @post = current_user.posts.find(params[:id])
   end
 
+  def create
+    post = current_user.posts.new(post_params)
+    post.save
+    redirect_to profile_posts_path
```
Comment (d-atkins) [inline app/controllers/profile/posts_controller.rb:14]: This should redirect to the posts index, double check the user story

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -6,3 +6,20 @@
     <p><%= post.body %></p>
     </section>
   <% end %>
+  <div class="row">
+  <div class="col-xs-12 col-sm-6 col-sm-offset-3">
+    <%= form_for(:static_pages, url: profile_posts_path, :html =>{:class => 'form-horizontal'}) do |f| %>
```
Comment (d-atkins) [inline app/views/posts/index.html.erb:11]: I'd like to go over this when you have time

**Review by d-atkins — CHANGES_REQUESTED** [review]
Missing a feature test - the user story walks through everything the feature test should be confirming

Does not work in development - the controller should be figuring out how to apply the tone and day, please go over the notes in the user story and reach out if you're confused on anything

**Comment by mronauli** [comment]
made the edits you suggested. veered in a slightly different direction than the find or create by method from the user story but ultimately i would say this works as a placeholder @d-atkins

File: app/controllers/profile/posts_controller.rb [hunk 4]
```diff
@@ -8,4 +8,19 @@ def show
     @post = current_user.posts.find(params[:id])
   end
 
+  def create
+    day = Day.create(week: 1, day_of_week: 1, group: current_user.group)
```
Comment (d-atkins) [inline app/controllers/profile/posts_controller.rb:12]:
```suggestion
    day = Day.find_or_create_by(week: 1, day_of_week: 1, group: current_user.group)
```
We don't want to create copies of days

**Review by d-atkins — CHANGES_REQUESTED** [review]

**Review by d-atkins — APPROVED** [review]

**Pull request merged**